src/mono/mono/metadata/threads.c

```diff
@@ -4096,15 +4061,6 @@ mono_thread_execute_interruption (MonoExceptionHandle *pexc)
 		/* calls UNLOCK_THREAD (thread) */
 		self_suspend_internal ();
 		unlock = FALSE;
-	} else if (MONO_HANDLE_GETVAL (thread, thread_interrupt_requested)) {
```

Comment:
I think when you get rid of this, you make `mono_sleep_internal` uninterruptible - it calls this function inside its sleep loop.